index.js

```diff
@@ -20,35 +22,293 @@ class J5sExecutor extends Executor {
+            if (exists) {
+                return this.breaker.runCommand({
+                    module: 'job',
+                    action: 'config',
+                    params: [{ name: jobName, xml }]
+                });
+            }
+
+            return this.breaker.runCommand({
+                module: 'job',
+                action: 'create',
+                params: [{ name: jobName, xml }]
+            });
+        });
+    }
+
+    /**
+     * Stop Jenkins job
+     * @method _jenkinsJobStop
+     * @param  {String}   jobName            Jenkins job name
+     * @return {Promise}
+     */
+    _jenkinsJobStop(jobName) {
+        return this.breaker.runCommand({
+            module: 'job',
+            action: 'get',
+            params: [{ name: jobName }]
+        }).then((data) => {
+            if (!(data && data.lastBuild && data.lastBuild.number)) {
+                throw new Error('No build has been started yet, try later');
+            }
+
+            return this.breaker.runCommand({
+                module: 'build',
+                action: 'stop',
+                params: [{ name: jobName, number: data.lastBuild.number }]
+            });
+        }).then(() => this._jenkinsJobWaitStop(jobName, 0));
+    }
+
+    /**
+     * Wait until the job has stopped
+     * @method _jenkinsJobWaitStop
+     * @param  {String}   jobName            Jenkins job name
+     * @param  {Number}   timeConsumed       Elapsed time
+     * @return {Promise}
+     */
+    _jenkinsJobWaitStop(jobName, timeConsumed) {
+        if (timeConsumed >= this.cleanupTimeLimit) {
+            return Promise.resolve();
+        }
+
+        return this.breaker.runCommand({
+            module: 'job',
+            action: 'get',
+            params: [{ name: jobName }]
+        }).then((data) => {
+            if (data && data.lastCompletedBuild && data.lastCompletedBuild.number) {
+                return null;
+            }
+
+            // delay between retry attempts
+            return new Promise((resolve) => {
+                setTimeout(() => resolve(), this.cleanupWatchInterval * 1000);
+            }).then(() =>
+                this._jenkinsJobWaitStop(jobName, timeConsumed + this.cleanupWatchInterval)
+            );
+        });
+    }
+
+    /**
+     * Jenkins job config xml
+     * @method _loadJobXml
+     * @param  {Object}   config        A configuration object psssed to _start
+     * @return {Promise}
+     */
+    _loadJobXml(config) {
+        const { buildScript, cleanupScript } = this._taskScript(config);
+
+        return new Promise((resolve, reject) => {
+            const configPath = path.resolve(__dirname, './config/job.xml');
+
+            fs.readFile(configPath, 'utf-8', (err, fileContents) => {
+                if (err) {
+                    return reject(err);
+                }
+
+                return resolve(fileContents);
+            });
+        }).then(xml => (
+            _.template(xml)({
```

Comment:
You've brought in the entirety of lodash just for this one method. That's about 1MB of code, most of which isn't used. Are there other, less heavy, options that might work? lodash-template (18k), or tinytim (16k)